File: hanoi.cpp
#include <iostream>

void hanoi(int n,char x,char y,char z){
  if(n>1) hanoi(n-1,x,z,y);
  std::cout<<x<<"=>"<<y<<std::endl;
  if(n>1) hanoi(n-1,z,y,x);
}

int main(){
  int n;
  std::cin>>n;
  hanoi(n,'a','b','c');
  return 0;
}

File: hanoi.dart
import 'dart:io';

void hanoi(int n, String x, String y, String z) {
  if (n > 1) hanoi(n - 1, x, z, y);
  stdout.write('$x=>$y\n');
  if (n > 1) hanoi(n - 1, z, y, x);
}

void main() {
  String n = stdin.readLineSync();
  hanoi(int.parse(n), 'a', 'b', 'c');
}
